pkg/virt-controller/watch/migration/migration.go

```diff
@@ -466,7 +474,7 @@ func (c *Controller) updateStatus(migration *virtv1.VirtualMachineInstanceMigrat
 		c.recorder.Eventf(migration, k8sv1.EventTypeWarning, controller.FailedMigrationReason, "Migration failed because vmi does not exist.")
 		log.Log.Object(migration).Error("vmi does not exist")
 	} else if vmi.IsFinal() {
-		err := c.failMigration(migrationCopy)
+		err := c.interruptMigration(migrationCopy, vmi)
```

Comment:
Have you considered handling this case only once, after `if migration.IsFinal() {` - I mean, let me migration fail and then check the status of the volumes.

Reply:
So we would check the volumes for every failed migration? While it would slightly reduce the footprint of this PR, I'm not sure I like the overhead and the fact that a successful migration could be marked as failed...

Reply:
I'm worried about the maintainability. I wonder if it will be easy to extend. I think there is a chance that people may get it wrong and will forget to call this function on next changes.  
I may be wrong, but it felt that calling it in one place is better since it won't be overlooked. I'm not sure if the overhead of checking every failed migration is that large. 

On the other hand, we can't call these `successful` migrations either. We can't be sure that we can always provide full the `migrationState` and even if the migration is running on the target we don't know if the target virt-handler called finalizeMigration()

Reply:
It's not a blocker. I'm just thinking out loud.

Reply:
@jean-edouard do you have any thoughts on this or do you prefer to keep it the way it is right now? we can improve this in a follow up too.

Reply:
Yeah I'd rather stick with how it currently is for now, this PR is getting old

Reply:
Ok, it's not critical right now. we can follow up.